source/cli-translate.js

```diff
@@ -0,0 +1,128 @@
+var imagemagick_cli_lists = null;
+
+function addSelectOptions(node)
+{
+    var selectCliLanguageNode = document.createElement("select");
+    selectCliLanguageNode.className += "btn text-dark bg-light mx-4";
+    selectCliLanguageNode.style.fontSize=".75rem";
+    selectCliLanguageNode.style.float="right";
+    selectCliLanguageNode.style.clear="both";
+    selectCliLanguageNode.style.position="relative";
+    selectCliLanguageNode.style.top="-18px";
+    selectCliLanguageNode.style.height="22px";
+    selectCliLanguageNode.style.padding="0px";
+    selectCliLanguageNode.onchange = function(){setCliChanged(this)};
+    selectCliLanguageNode.title = "Change Operating System Syntax";
+    var selectCliLanguageOptions = ["Linux","Windows","Bat File"];
+
+    for(let option of selectCliLanguageOptions){
+        var optionNode = document.createElement("option");
+        optionNode.text=option;
+        selectCliLanguageNode.appendChild(optionNode);
+    }
+    node.after(selectCliLanguageNode);
+    return selectCliLanguageNode;
+}
+
+const imagemagick_linux_to_windows_replacements = new Map([
+    [/\\\s*\n/gm, "^\n"], //replace newlines with ^\n
+    [/\s?\\(\(|\))\s?/gm, " $1 "], // unesecape parens, add space on either side to ensure compatibility "word\)" is valid in linux in windows "word)" is not.
+    [/\\!/gm,"!"], // unescape exclamation point.
+    [/'/gm, "\""], //single quote to double quote, breaks on nested quotes.
+    [/\&|\||\<|\>/gm, "\\$&"], //Escape &,|,<,> - note: appears unused.
+]);
+
+const imagemagick_linux_to_windows_bat_replacements = new Map([
+    [/%/gm, "%%"], //Double up % signs. Needed for bat files, but not for CMD.
+    [/^\s*#/gm, "::"], //Replace UNIX comment # with two colons. No support for inline comments.
+])
+
+function translateLinuxToWindows(node) {
+    imagemagick_linux_to_windows_replacements.forEach(function(value,key){
+        node.innerHTML = node.innerHTML.replaceAll(key,value);
+    });
+}
+
+function translateWindowsToBatch(node) {
+    imagemagick_linux_to_windows_bat_replacements.forEach(function(value,key){
+        node.innerHTML = node.innerHTML.replaceAll(key,value);
+    });
+}
+
+function addLinuxWindowsCLIToggle(node) {
+    var copyNode = node.cloneNode(true);
+    copyNode.classList.add("cli-windows");
+    copyNode.style.display = "none";
+    translateLinuxToWindows(copyNode);
+    var batchNode = copyNode.cloneNode(true);
+    batchNode.classList.add("cli-batch");
+    batchNode.style.display = "none";
+    translateWindowsToBatch(batchNode);
+    if(batchNode.innerHTML == node.innerHTML) // No change, ignore.
+    {
+        copyNode.remove();
+        return [];
+    }
+
+    node.classList.add("cli-linux");
+    node.after(copyNode);
+    copyNode.after(batchNode);
+    return [copyNode,batchNode];
+}
+
+function addCliTranslations() {
```

Comment:
This is done in one synchronous block. I did not notice any loading speed issues with this approach. It would be possible to redesign this system to do so more lazily, or over a series of frames, but I don't personally think it is necessary. (it might be more necessary in the imagemagick.org/usage examples since there are many many more examples per page there.)

Reply:
I also tested this locally and this seems to run okay. We can always add a more lazy approach in the future.